luigi/contrib/bigquery.py

```diff
@@ -282,9 +282,9 @@ def copy(self,
 
 
 class BigqueryTarget(luigi.target.Target):
-    def __init__(self, project_id, dataset_id, table_id, client=None, tmp_dataset_id=None):
+    def __init__(self, project_id, dataset_id, table_id, client=None, tmp_dataset_id=None, tmp_salt=None):
         self.table = BQTable(project_id=project_id, dataset_id=dataset_id, table_id=table_id)
-        tmp_table_id = "_" + table_id + "_%09d" % random.randrange(0, 1e10)
+        tmp_table_id = "_" + table_id + (tmp_salt or "_%09d" % random.randrange(0, 1e10))
```

Comment:
Hmm since most tasks that use BQ directly don't need this, what do you think about having an interface like:

```python
target = BigqueryTarget(...)
temp_table = target.temp_table(salt=...)
# temp_table is a BQTable
DO_HADOOP_CONNECTOR_JOB()
```

It might be nice to do it as a context manager as well, but I think with the way luigi classes are usually implemented, you might need the temp table name outside of the run() method - it'll get weird.

Separately, I'd suggest using a sha hash of the table id or something - using a non-consistent table name might lead to subtle bugs (i.e. two BigqueryTarget() instances pointing to the same project/dataset/table don't have the same temp table id)